src/auth/baseexternalclient.ts

```diff
@@ -39,6 +39,8 @@ const STS_REQUEST_TOKEN_TYPE = 'urn:ietf:params:oauth:token-type:access_token';
 const DEFAULT_OAUTH_SCOPE = 'https://www.googleapis.com/auth/cloud-platform';
 /** The google apis domain pattern. */
 const GOOGLE_APIS_DOMAIN_PATTERN = '\\.googleapis\\.com$';
+/** */
+const STAR_PATTERN = '[^\\.\\s\\/\\\\]';
```

Comment:
You can include the `+` too, since it is also include in all the patterns:
`[^\\.\\s\\/\\\\]+`